satpy/readers/lmi_l2.py

```diff
@@ -0,0 +1,154 @@
+                # we need to omit file_key once.
+                handled_variables.add(var_name)
+                new_info = ds_info.copy()  # don't mess up the above yielded
+                yield True, new_info
+            elif is_avail is None:
+                # if we didn't know how to handle this dataset
+                # and no one else did,
+                # then we should keep it going down the chain.
+                yield is_avail, ds_info
+
+        # Iterate over dataset contents
+        for var_name, val in self.file_content.items():
+            # Only evaluate variables
+            if isinstance(val, netCDF4.Variable):
+                logger.debug("Evaluating new variable: %s", var_name)
+                var_shape = self[var_name + "/shape"]
+                logger.debug("Dims:{}".format(var_shape))
+                if (lat_shape == var_shape):
+                    logger.debug("Found valid additional dataset: %s",
+                                 var_name)
+
+                    # Skip anything we have already configured
+                    if (var_name in handled_variables):
+                        logger.debug("Already handled, skipping: %s", var_name)
+                        continue
+                    handled_variables.add(var_name)
+                    logger.debug("Using short name of: %s", var_name)
+
+                    # Create new ds_info object
+                    new_info = {
+                        'name': var_name,
+                        'file_key': var_name,
+                        'coordinates': ['LON', 'LAT'],
+                        'file_type': self.filetype_info['file_type'],
+                        'resolution': self.filename_info['resolution'].lower()
+                    }
+                    yield True, new_info
+
+    def get_metadata(self, data, ds_info):
+        """Get metadata."""
+        metadata = {}
+        metadata.update(data.attrs)
+        units = data.attrs['units']
+        # fix the wrong unit "uJ/m*m/ster"
+        if not units.isascii():
+            metadata['units'] = b'\xc2\xb5J/m*m/ster'
+        metadata.update(ds_info)
+        metadata.update({
+            'platform_shortname': self.filename_info['platform_id'],
+            'sensor': self.filename_info['instrument'].lower(),
+            'start_time': self.start_time,
+            'end_time': self.end_time,
+        })
+
+        return metadata
+
+    def get_dataset(self, ds_id, ds_info):
+        """Load a dataset."""
+        logger.debug("Getting data for: %s", ds_id.name)
+        file_key = ds_info.get('file_key', ds_id.name)
+        data = self[file_key]
+        data.attrs = self.get_metadata(data, ds_info)
+        # rename coords
+        data = data.rename({'x': 'y'})
+        # assign 'y' coords which is useful for multiscene,
+        # although the units isn't meters
+        len_data = data.coords['y'].shape[0]
+        data.coords['y'] = np.arange(len_data)
```

Comment:
Why do you need to assign these? I don't think you need `y` coordinates here. Maybe I'm wrong and naming it `y` is a bad idea, but I think that is consistent with other readers and the rest of Satpy. The dimension should be `y`, but no `coords` for it. My thought was instead that you add a `data.coords['time'] = [... series of times ...]` then the MultiScene `timeseries` function could be updated to look at that coordinate. So `time` would be the only coordinate you add in the reader, the base reader will then add `lon` and `lat` for you.

Reply:
@djhoese 

Got it! Here's the method:
```
        tseries = pd.DatetimeIndex(np.repeat(data.attrs['start_time'], data.shape[0]))
        data = data.expand_dims({'time': tseries})
        data = data.rename({'x': 'y'})
```
Example:
```
<xarray.DataArray 'ER' (time: 31, y: 31)>
dask.array<where, shape=(31, 31), dtype=float32, chunksize=(31, 31), chunktype=numpy.ndarray>
Coordinates:
  * time     (time) datetime64[ns] 2019-07-25T05:25:10 ... 2019-07-25T05:25:10
Dimensions without coordinates: y
```
But, I will get this error when using `MultiScene` by `mscn = MultiScene.from_files(filenames, reader='lmi_l2', group_keys=('start_time', 'subtask_num'))` and `imgs = mscn.blend(blend_function=timeseries)`:
```
ValueError: arguments without labels along dimension 'y' cannot be aligned because they have different dimension sizes: {8, 74, 44, 45, 86, 55, 24, 60, 93, 31}
```

The modification of `MultiScene` is mentioned in #1173.